boards/nordic/nrf9280pdk/nrf9280pdk_nrf9280-memory_map.dtsi

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/dt-bindings/memory-attr/memory-attr.h>
+
+/ {
+	reserved-memory {
+		cpuapp_ram0x_region: memory@2f011000 {
+			compatible = "nordic,owned-memory";
+			reg = <0x2f011000 DT_SIZE_K(260)>;
+			status = "disabled";
+			perm-read;
+			perm-write;
+			perm-secure;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0x2f011000 0x41000>;
+
+			cpusec_cpuapp_ipc_shm: memory@0 {
+				reg = <0x0 DT_SIZE_K(2)>;
+			};
+
+			cpuapp_cpusec_ipc_shm: memory@800 {
+				reg = <0x800 DT_SIZE_K(2)>;
+			};
+
+			cpuapp_data: memory@1000 {
+				reg = <0x1000 DT_SIZE_K(256)>;
+			};
+		};
+
+		cpurad_ram0x_region: memory@2f052000 {
+			compatible = "nordic,owned-memory";
+			reg = <0x2f052000 DT_SIZE_K(4)>;
+			status = "disabled";
+			perm-read;
+			perm-write;
+			perm-secure;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0x2f052000 0x1000>;
+
+			cpusec_cpurad_ipc_shm: memory@0 {
+				reg = <0x0 DT_SIZE_K(2)>;
+			};
+
+			cpurad_cpusec_ipc_shm: memory@800 {
+				reg = <0x800 DT_SIZE_K(2)>;
+			};
+		};
+
+		cpuapp_cpurad_ram0x_region: memory@2f0bf000 {
+			compatible = "nordic,owned-memory";
+			reg = <0x2f0bf000 DT_SIZE_K(4)>;
+			status = "disabled";
+			perm-read;
+			perm-write;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0x2f0bf000 0x1000>;
+
+			cpuapp_cpurad_ipc_shm: memory@0 {
+				reg = <0x0 DT_SIZE_K(2)>;
+			};
+
+			cpurad_cpuapp_ipc_shm: memory@800 {
+				reg = <0x800 DT_SIZE_K(2)>;
+			};
+		};
+
+		cpuapp_cpumod_ram0x_region: memory@2f0c0000 {
+			compatible = "nordic,owned-memory";
+			reg = <0x2f0c0000 DT_SIZE_K(20)>;
+			status = "disabled";
+			perm-read;
+			perm-write;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0x2f0c0000 0x5000>;
+
+			/* Define another node with ownder-id 4 to allocate
+			 * this area and share it with the modem code.
+			 */
```

Comment:
This structure (reserved-memory/nordic,owned-memory/nordic,owned-memory) looks confusing.

I'm thinking about something a little different like:

```
reserved-memory {
    cpuapp_cpumod_ram0x_region: memory@2f0c0000 {
        reg = <0x2f0c0000 DT_SIZE_K(20)>;
        #address-cells = <1>;
        #size-cells = <1>;
        ranges = <0x0 0x2f0c0000 DT_SIZE_K(20)>;

        /* Memory partitions */
        cpuapp_cpumod_ipc_shm_ctrl: memory@0 {
            reg = <0x0 0xce0>;
        };
        /* TX heap, user defined */
        cpuapp_cpumod_ipc_shm_heap: memory@ce0 {
            reg = <0xce0 0x2080>;
        };
        /* RX heap, user defined */
        cpumod_cpuapp_ipc_shm_heap: memory@2d60 {
            reg = <0x2d60 0x2000>;
        };

        /* Shared memory ownership */
        cpuapp_cpumod_ipc_shm: memory@2 {
            compatible = "nordic,owned-memory";
            reg = <0x0 DT_SIZE_K(20)>;
            owner-id = <2>;
            perm-read;
            perm-write;
            status = "disabled";
        };
        cpumod_cpuapp_ipc_shm: memory@4 {
            compatible = "nordic,owned-memory";
            reg = <0x0 DT_SIZE_K(20)>;
            owner-id = <4>;
            perm-read;
            perm-write;
            status = "disabled";
        };
    };
};
```

What do you think?

Reply:
BTW, it would be perfect for this case if `owner-id` could be a list like `owner-id = <2 4>;`. But nrf-regtool doesn't support this. Or does it?

Reply:
> BTW, it would be perfect for this case if `owner-id` could be a list like `owner-id = <2 4>;`. But nrf-regtool doesn't support this. Or does it?

The intended solution was to use the `nordic,access` property from #72273. At first, it wasn't supposed to be merged until the relevant secure domain work was completed.

We could start using it now though, but only for the multi-owner use case, and without deprecating the `perm-*` properties yet. I can update the PR if that sounds reasonable.

Reply:
Right, I forgot about `nordic,access`. That's the correct solution. For me, it's acceptable to keep (reserved-memory/nordic,owned-memory/nordic,owned-memory) temporarily. Just add a TODO comment to refactor it once #72273 is merged

Reply:
I went with your suggestion here, and left a TODO.